### ISSUE TITLE
Can't arm, but disarming works fine.

### ISSUE THREAD

**Issue by lonelydime** [pakerfeldt/node-red-contrib-envisalink#7]
Hopefully I'm not missing something completely obvious,

I can sent the string "0401####" using a user code to disarm the system if it's armed, but I cannot get the system to arm using "0311" for stay or "0331####" using either the master code or a user code.

Using 0311 as the payload, I get the warning "Code is Required - "
Using 0331#### I don't get any response, but the system does nothing.

I'm using a DSC1864 and Envisalink 3.  Any help would be greatly appreciated.

**Comment by pakerfeldt** [comment]
That's weird. Can you arm away by sending 0301? Also 0311 shouldn't require code. Just like for 0301.

**Comment by lonelydime** [comment]
Same response with 0301, it tells me that a code is required.  Here are some screens of my test setup:
https://imgur.com/a/gC5XTCd

Previously I was running OpenHAB2 which has it's own set of issues, but seems to be able to arm stay and disarm so it's probably something setup related I'm doing, I'm just banging my head against a wall trying to figure it out.  I do a similar command with the OpenHAB DSC binding:
sendCommand(SEND_DSC_ALARM_COMMAND, "031,1")

I've tried sending a code along with 0301 and 0311 which does nothing, which it shouldn't, but I figured I'd give it a shot.  Thanks for the response!

**Comment by pakerfeldt** [comment]
Looks like your envisalink responds with 900 after you try to arm. Not entirely sure why, I can debug later but the next few days I don't have access to my computer. I assume you're running the latest version?

Also, in the meantime you can try arming using raw input: 0711xxxx where xxxx is your code to arm. That would essentially emulate someone pressing those buttons on your keypad.

**Comment by Gnindit** [comment]
I just tried two buttons in the UI.
One with payload string 0301 and topic emtpy.
the other one with payload string 0401#### and topic emtpy.
 And they arm and disarm for me.

**Comment by lonelydime** [comment]
I tried clearing out my topics, it still requires a code.  Thanks for the suggestion though!

My Envisalink3's firmware version is 01.07.88, I just upgraded to 01.12.167 with the same result (requesting code to arm on 0311 and 0301).  I'm running node-red-contrib-envisalink 1.1.5, npm 6.4.1, node red version 0.20.7, and node 8.16 in a docker container.

Trying the arm function through eyez-on.com results in the same asking for a code on the envisalink's local page.

Arming with 0711xxxx works, thanks for the work around!

**Comment by lonelydime** [comment]
After more tinkering, this very well may be an option with the DSC powerseries panels.  I did a self install and set the options myself, so I'm not sure what a standard installation looks like.  I got it to arm without asking for a code by enabling the quick arming feature of the panel:
[015] Third System Option Code
    [4] ON: the Quick Arming [ * ][0] feature will be enabled. OFF: Quick Arming [ * ][0] feature will be disabled.
    * If this feature is disabled, a valid user code must be entered after the Stay or Away function buttons are pressed.

The OpenHAB DSC binding stores the user code in a setting and very well may take this setting into account when sending the 031 code which is why I've never had a problem before.

After enabling this feature I can now send the 031 code and it'll arm without problem.  Before I had the away and stay arming buttons require a code, which isn't a big deal to remove.

Thanks for the help everyone!
